source/JsMaterialX/README.md

```diff
@@ -0,0 +1,88 @@
+# Javascript Support
+
+A Javascript package is created from the following modules.
+
+- [JsMaterialXCore](JsMaterialXCore): Contains all of the core classes and util functions.
+- [JsMaterialXFormat](JsMaterialXFormat): Contains the `readFromXmlString` function to read a MaterialX string.
+
+## Generating JS/WASM
+
+### Prerequisites
+
+Make sure to clone the [emsdk repository](https://github.com/emscripten-core/emsdk), install and enable the latest emsdk environment.
+```sh
+# Get the emsdk repo
+git clone https://github.com/emscripten-core/emsdk.git
+
+# Enter that directory
+cd emsdk
+
+# Download and install the latest SDK tools.
+./emsdk install latest
+
+# Make the "latest" SDK "active" for the current user. (writes ~/.emscripten file)
+./emsdk activate latest
+```
+
+For more information follow the steps described in the [emscripten documentation](https://emscripten.org/docs/getting_started/downloads.html). 
+
+### Build
+In the root of directory of this repository run the following:
+
+#### CMake
+The JavasScript library can be built using cmake and make.
+
+1. Activate the emscripten environment
+```sh
+source ../../../../emsdk/emsdk_env.sh
+```
+
+2. Create the `_build` folder
+```sh
+mkdir -p ./_build
+cd ./_build
+```
+
+3. Run cmake and make
+```sh
+# This will generate the release library
+# To build the non minified debug version replace -DMATERIALX_JS_BUILD_TYPE=RELEASE with -DMATERIALX_JS_BUILD_TYPE=DEBUG
+emcmake cmake ../../.. -DMATERIALX_BUILD_JS=ON -DMATERIALX_JS_BUILD_TYPE=RELEASE
+emmake make
+```
+
+#### build.sh
+There is a helper script called `build.sh` that will do the above steps.
+
+1. Build the release WebAssembly binary and the JavaScript library.
+
+```sh
+./build.sh
+```
+
+To build the debug version call:
+```sh
+./build.sh DEBUG
+```
+
+### Output
+After building the project the `MaterialXLib.wasm` and `MaterialXLib.js` files can be found in `./_build/source/JsMaterialX/`.
+
+### Testing
+The JavaScript tests are located in `./test` folder and are defined with the `.spec.js` suffix.
+Most of these tests were copied over from the Python [main.py tests](../../python/MaterialXTest/main.py).
+
+#### Setup
+These tests require node.js and npm which can be installed [HERE](https://nodejs.org/en/download/).
+
+1. Install the npm packages.
+```sh
+cd ./test && npm install
```

Comment:
This does not seem to work for me on Windows with the following error:
```
npm ERR! file d:\Work\materialx\MaterialX-adsk-dev\source\JsMaterialX\test\package.json
npm ERR! code EJSONPARSE
npm ERR! JSON.parse Failed to parse json
npm ERR! JSON.parse Unexpected token } in JSON at position 277 while parsing near '...babel/register",
npm ERR! JSON.parse   },
npm ERR! JSON.parse   "keywords": []...'
npm ERR! JSON.parse Failed to parse package.json data.
npm ERR! JSON.parse package.json must be actual JSON, not just JavaScript.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\kwokb\AppData\Roaming\npm-cache\_logs\2020-05-21T19_52_00_992Z-debug.log
```
This is the full log:
```
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'd:\\Work\\materialx\\emsdk\\node\\12.9.1_64bit\\bin\\node.exe',
1 verbose cli   'd:\\Work\\materialx\\emsdk\\node\\12.9.1_64bit\\bin\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'install'
1 verbose cli ]
2 info using npm@6.10.2
3 info using node@v12.9.1
4 verbose npm-session f249194f73d37e2d
5 silly install runPreinstallTopLevelLifecycles
6 silly install loadCurrentTree
7 silly install readLocalPackageData
8 timing stage:rollbackFailedOptional Completed in 3ms
9 timing stage:runTopLevelLifecycles Completed in 16ms
10 verbose stack Error: Failed to parse json
10 verbose stack Unexpected token } in JSON at position 277 while parsing near '...babel/register",
10 verbose stack   },
10 verbose stack   "keywords": []...'
10 verbose stack     at parseError (d:\Work\materialx\emsdk\node\12.9.1_64bit\bin\node_modules\npm\node_modules\read-package-json\read-json.js:452:11)
10 verbose stack     at parseJson (d:\Work\materialx\emsdk\node\12.9.1_64bit\bin\node_modules\npm\node_modules\read-package-json\read-json.js:104:26)
10 verbose stack     at d:\Work\materialx\emsdk\node\12.9.1_64bit\bin\node_modules\npm\node_modules\read-package-json\read-json.js:51:5
10 verbose stack     at d:\Work\materialx\emsdk\node\12.9.1_64bit\bin\node_modules\npm\node_modules\graceful-fs\graceful-fs.js:90:16
10 verbose stack     at FSReqCallback.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:61:3)
11 verbose cwd d:\Work\materialx\MaterialX-adsk-dev\source\JsMaterialX\test
12 verbose Windows_NT 10.0.18363
13 verbose argv "d:\\Work\\materialx\\emsdk\\node\\12.9.1_64bit\\bin\\node.exe" "d:\\Work\\materialx\\emsdk\\node\\12.9.1_64bit\\bin\\node_modules\\npm\\bin\\npm-cli.js" "install"
14 verbose node v12.9.1
15 verbose npm  v6.10.2
16 error file d:\Work\materialx\MaterialX-adsk-dev\source\JsMaterialX\test\package.json
17 error code EJSONPARSE
18 error JSON.parse Failed to parse json
18 error JSON.parse Unexpected token } in JSON at position 277 while parsing near '...babel/register",
18 error JSON.parse   },
18 error JSON.parse   "keywords": []...'
19 error JSON.parse Failed to parse package.json data.
19 error JSON.parse package.json must be actual JSON, not just JavaScript.
20 verbose exit [ 1, true ]
```

Reply:
This was related to the typo in package.json